### PR TITLE
Issue/2752 Add Dataset Review Page

### LINKED ISSUE #2752: Add "Review" page to the end of the Add Dataset flow
## Description

It's important for both correctness and accessibility that a data steward has the opportunity to review what they've submitted before either publishing a new dataset, or sending it off for review.

To allow for this, we need a Review Page at the end of the Add Dataset flow.

See collapsed design here: https://projects.invisionapp.com/share/KMVQ7DH652J#/screens/402485648

and expanded design on next screen here: https://projects.invisionapp.com/share/KMVQ7DH652J#/screens/407556243

This will display a section for each page of the Add Dataset flow, showing the most important fields from each page, along with an "expand" button that allows the user to view the rest of the values that have been input, without leaving the page. Each section will also have an "Edit" button which will return the user to that page so that they can make edits, before going back to the review page.

## Acceptance Criteria
- After entering all the details of a dataset, the user lands on the Review page
- The review page has a section for every page in Add Dataset, showing a subset of fields per page as per the design.
    - These fields are all read-only
    - The fields that display when the section is in a collapsed state are the fields that are mandatory
- Clicking on the expand (+) button on any section will expand it, showing all fields on that page, but read-only (there's no designs for this, just make it look like the rest of the read-only fields on the page
     - When a section is in an expanded state, the expand button should turn into a contract (-) button that does the opposite (shrinks the section back to its original state)
- Clicking on the "Edit" button for a section returns the user to the page relating to the section in question
    - When the user has gone back to a section in this way, the primary button at the bottom of the page should now say "Return to Review", and when pressed, take the user back to the review page.
    - The other two buttons should still be present and clickable, but they should not have a primary style.
    - The "Return to Review" button should trigger validation, and should not allow the user to return to the review until they've successfully entered valid input for the page.
- Contrary to the designs, the "approver note" page doesn't need an expand/contract button

## Technical Notes
- The "edit" button could probably be achieved by simply having a link back to each page with some kind of query param that flags that the user should be allowed to return to the review page

### PR DESCRIPTION
**Pull request by t83714**
### What this PR does

Fixes #2752 
- Adjusted progress meter & workflow
- Minor changes to search API so I can pull region name by region id
- Minor logic adjust to some input control & data population logic

### test site:

https://issue-2752.dev.magda.io

As it involves backend changes, it's deployed as no-data preview site.

### Checklist

-   [x] There are unit tests to verify my changes are correct
-   [x] I've updated CHANGES.md with what I changed.
-   [x] I've linked this PR to an issue in ZenHub (core dev team only)

### CODE REVIEW

**Comment by t83714** [comment]
It's ready to review now~

File: magda-web-client/src/Components/Dataset/Add/Pages/ReviewPage/AccessAndUse.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+                    <div className="title-box">
+                        Who can see the dataset once it is published
+                        <ValidationRequiredLabel validationFieldPath="$.datasetPublishing.level" />
+                        ?
+                    </div>
+                </div>
+                <div className="col-sm-9 content-box single-line">
+                    {datasetPublishing?.level
+                        ? codelists.publishingLevel[datasetPublishing.level]
+                        : "N/A"}
```
Comment (AlexGilleran) [inline magda-web-client/src/Components/Dataset/Add/Pages/ReviewPage/AccessAndUse.tsx:65]: Can we pull this into a constant ( `NO_VALUE_LABEL` ?) so we can change the copy for all of them at once easily if we want to?

File: magda-web-client/src/Components/Dataset/Add/Pages/ReviewPage/AccessAndUse.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+            {config.featureFlags.publishToDga ? (
+                <CollapseItem className="row" alwaysShow={true}>
+                    <div className="col-sm-3">
+                        <div className="title-box">
+                            Publish to data.gov.au as open data (*):
+                        </div>
+                    </div>
+                    <div className="col-sm-9 content-box single-line">
+                        {datasetPublishing?.publishAsOpenData?.["dga"]
+                            ? "YES"
```
Comment (AlexGilleran) [inline magda-web-client/src/Components/Dataset/Add/Pages/ReviewPage/AccessAndUse.tsx:45]: Can this be `"Yes" : "No"` so it's a bit less shouty? :)

File: magda-web-client/src/Components/Dataset/Add/Pages/ReviewPage/CollapseBox.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                    }
+                >
+                    {isNotCollapsible ? null : (
+                        <img
+                            src={isOpen ? minusIcon : plusIcon}
+                            onClick={onToggleClick}
+                        />
+                    )}
+
+                    <button
```
Comment (AlexGilleran) [inline magda-web-client/src/Components/Dataset/Add/Pages/ReviewPage/CollapseBox.tsx:52]: Is it possible to make this a `<Link>` so you can hover over it to see where it goes / open in new tab / etc

File: magda-web-client/src/Components/Dataset/Add/Pages/ReviewPage/AccessAndUse.tsx (hunk 1 above)
Comment (t83714) [reply magda-web-client/src/Components/Dataset/Add/Pages/ReviewPage/AccessAndUse.tsx]: Updated

File: magda-web-client/src/Components/Dataset/Add/Pages/ReviewPage/AccessAndUse.tsx (hunk 2 above)
Comment (t83714) [reply magda-web-client/src/Components/Dataset/Add/Pages/ReviewPage/AccessAndUse.tsx]: Updated~

File: magda-web-client/src/Components/Dataset/Add/Pages/ReviewPage/CollapseBox.tsx (hunk 3 above)
Comment (t83714) [reply magda-web-client/src/Components/Dataset/Add/Pages/ReviewPage/CollapseBox.tsx]: Changed to <Link>

**Comment by t83714** [comment]
@AlexGilleran I think it's ready for another look ~ Please let me know anything needs changes when you have time~

**Pull request merged**